src/data/equiv/basic.lean

```diff
@@ -1498,10 +1498,18 @@ equiv.ext (λ x, begin
   split_ifs; simp
 end)
 
-@[simp] lemma swap_apply_self {α : Type*} [decidable_eq α] (i j a : α) :
+@[simp] lemma swap_apply_self (i j a : α) :
   swap i j (swap i j a) = a :=
 by rw [← equiv.trans_apply, equiv.swap_swap, equiv.refl_apply]
 
+/-- A function is invariant to a swap if it is equal at both elements -/
+lemma swap_invariant {v : α → β} {i j : α} (hv : v i = v j) (k : α) : v (swap i j k) = v k :=
```

Comment:
Looks like `invariant` isn't a commonly used term in the naming conventions, so let's stick to the existing name parts.
```suggestion
lemma apply_swap_eq_self {v : α → β} {i j : α} (hv : v i = v j) (k : α) : v (swap i j k) = v k :=
```

Reply:
Sure, seems reasonable.